File: backend/src/main/java/com/example/bookhub/dto/BookListDto.java
package com.example.bookhub.dto;

import com.example.bookhub.enums.Genre;
import com.example.bookhub.enums.Language;
import lombok.Data;

import java.util.List;

@Data
public class BookListDto {

    private Long id;

    private String title;

    private String author;

    private Genre genre;

    private Language language;

    private double averageRating;

    private int numberOfRatings;

    private String description;

    private String imagePath;

    private List<ShelfDto> shelves;
}

File: frontend/src/app/core/services/book/models/book-list.dto.ts
import { Genre } from '../../../../shared/enums/genre';
import { Language } from '../../../../shared/enums/language';
import { ShelfDto } from '../../../../shared/models/shelf.dto';

export interface BookListDto {
  readonly id: number;
  readonly title: string;
  readonly author: string;
  readonly genre: Genre;
  readonly language: Language;
  readonly averageRating: number;
  readonly numberOfRatings: number;
  readonly description: string;
  readonly imagePath: string;
  readonly shelves: ShelfDto[];
}
